### PR TITLE
refactor(lint): remove eslint errors

### PR DESCRIPTION
**Pull request by chethanbellechalu-intel**
Refactor(lint): Remove eslint errors

### CODE REVIEW

File: src/reactjs/components/DomainEditor/DomainEditor.tsx [hunk 1]
```diff
@@ -2,165 +2,164 @@
...
+      })
     }
-    this.showNotification();
-  };
+    this.showNotification()
+  }
 
-  showNotification = () =>
+  showNotification = (): any =>
     setTimeout(() => {
```
Comment (rsdmike) [inline src/reactjs/components/DomainEditor/DomainEditor.tsx:94]: this should be void?

File: src/reactjs/components/DeviceGrid/DeviceGrid.tsx [hunk 2]
```diff
@@ -6,193 +6,169 @@
...
             })
-        })
-            .catch(error => { this.gridApi.showNoRowsOverlay() })
-    };
+          })
 
-    onSelectionChanged = () => {
-        if (isFunc(this.props.getSelectedDevices)) {
-            this.props.getSelectedDevices(this.gridApi.getSelectedRows())
+          // let filteredList = this.props.selectedDevices.filter(device => this.sta)
```
Comment (rsdmike) [inline src/reactjs/components/DeviceGrid/DeviceGrid.tsx:131]: remove comment

File: src/reactjs/components/CIRAEditor/CiraEditor.tsx [hunk 3]
```diff
@@ -2,138 +2,138 @@
...
-      });
+        type: 'error'
+      })
     }
-    this.showNotification();
+    this.showNotification()
   }
 
-  showNotification = () => setTimeout(() => {
+  showNotification = (): any => setTimeout(() => {
```
Comment (rsdmike) [inline src/reactjs/components/CIRAEditor/CiraEditor.tsx:94]: this should be void?

File: src/reactjs/components/CIRAEditor/CiraGrid.tsx [hunk 4]
```diff
@@ -45,65 +45,65 @@ export class CiraGrid extends React.Component<CiraGridProps, CiraGridStates> {
...
-  };
+  }
 
-  //Fetch the CIRA config scripts to be displayed on the grid
-  fetchCiraConfigs = async () =>
+  // Fetch the CIRA config scripts to be displayed on the grid
+  fetchCiraConfigs = async (): Promise<any> =>
     await HttpClient.get(
-      `${this.props.rpsServer}/api/v1/admin/ciraconfigs`, this.props.rpsKey
+      `${String(this.props.rpsServer)}/api/v1/admin/ciraconfigs`, this.props.rpsKey
```
Comment (rsdmike) [inline src/reactjs/components/CIRAEditor/CiraGrid.tsx:86]: Why don't we just make rpsServer a string?

File: src/reactjs/components/DeviceGrid/DeviceGrid.tsx [hunk 5]
```diff
@@ -6,193 +6,169 @@
...
+  softSelectedDevice: any
 }
 
 export interface gridProps {
-    deviceId: string;
-    mpsServer: string;
-    getSelectedDevices?: any;
-    selectedDevices?: any;
-    filter?: string;
+  deviceId: any
```
Comment (rsdmike) [inline src/reactjs/components/DeviceGrid/DeviceGrid.tsx:34]: why were these changed to any?

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/core/Utilities/MouseHelper.ts [hunk 6]
```diff
@@ -3,110 +3,109 @@
...
+    this.parent.buttonmask &= (0xFFFF - (1 << e.button))
+    return this.mousemove(e)
   }
-  mousemove(e: React.MouseEvent) {
-    if (this.parent.state != 4) return true;
-    let pos = this.getPositionOfControl(this.parent.canvasControl);
-    this.parent.lastMouseX = (e.pageX - pos[0]) * (this.parent.canvasControl.height / this.parent.canvasControl.offsetHeight);
-    this.parent.lastMouseY = ((e.pageY - pos[1] + (this.parent.scrolldiv ? this.parent.scrolldiv.scrollTop : 0)) * (this.parent.canvasControl.width / this.parent.canvasControl.offsetWidth));
 
+  mousemove (e: React.MouseEvent): any {
```
[the remark this answers is not in the document]
Comment (chethanbellechalu-intel) [reply src/core/Utilities/MouseHelper.ts]: Updated the return type to boolean

File: src/core/Utilities/MouseHelper.ts [hunk 7]
```diff
@@ -3,110 +3,109 @@
...
     this.lastEvent = null
   }
 
-  GrabMouseInput() {
-    if (this.MouseInputGrab == true) return;
-    let c = this.parent.canvasCtx.canvas;
+  GrabMouseInput (): any {
+    if (this.MouseInputGrab) return
+    // const c = this.parent.canvasCtx.canvas
     // c.onmouseup = this.mouseup;
```
[the remark this answers is not in the document]
Comment (chethanbellechalu-intel) [reply src/core/Utilities/MouseHelper.ts]: Updated

File: src/reactjs/components/AuditLog/CustomHeader.tsx [hunk 8]
```diff
@@ -6,76 +6,75 @@
...
+import * as React from 'react'
+import { Tooltip } from '../shared/Tooltip'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 
 export interface HeaderParams {
-    displayName: any,
-    description: any,
-    enableSorting: any
+  displayName: string
+  description: any
```
[the remark this answers is not in the document]
Comment (chethanbellechalu-intel) [reply src/reactjs/components/AuditLog/CustomHeader.tsx]: Changed the types of description and enableSorting to string and boolean respectively

File: src/reactjs/components/DomainEditor/DomainEditor.tsx (hunk 1 above)
Comment (chethanbellechalu-intel) [reply src/reactjs/components/DomainEditor/DomainEditor.tsx:95]: Since setTimeout function returns a value, the return type has been updated to match the return type of setTimeout function.

File: src/reactjs/components/DeviceGrid/DeviceGrid.tsx (hunk 2 above)
Comment (chethanbellechalu-intel) [reply src/reactjs/components/DeviceGrid/DeviceGrid.tsx]: Updated

File: src/reactjs/components/CIRAEditor/CiraEditor.tsx (hunk 3 above)
Comment (chethanbellechalu-intel) [reply src/reactjs/components/CIRAEditor/CiraEditor.tsx]: Since setTimeout function returns a value, the return type has been updated to match the return type of setTimeout function.

File: src/reactjs/components/CIRAEditor/CiraGrid.tsx (hunk 4 above)
Comment (chethanbellechalu-intel) [reply src/reactjs/components/CIRAEditor/CiraGrid.tsx]: Updated with proper null checks for prop values taken from the application URL.

File: src/reactjs/components/DeviceGrid/DeviceGrid.tsx (hunk 5 above)
Comment (chethanbellechalu-intel) [reply src/reactjs/components/DeviceGrid/DeviceGrid.tsx]: Since deviceId and mpsServer values are taken from the URL while using webpack, the probable types are string and null. Updated now with proper null checks for all the files

**Review by rsdmike — APPROVED** [review]

**Review by rsdmike — APPROVED** [review]
